server/src/test/java/io/crate/expression/scalar/SubscriptObjectFunctionTest.java

```diff
@@ -34,13 +36,31 @@
     public void testEvaluate() throws Exception {
         assertEvaluate("subscript_obj({x=10}, 'x')", 10);
         assertEvaluate("subscript_obj(subscript_obj({x={y=10}}, 'x'), 'y')", 10);
+
+        sqlExpressions.setErrorOnUnknownObjectKey(true);
+        Asserts.assertThrowsMatches(
+            () -> assertEvaluate("subscript_obj(subscript_obj({x={y=10}}, 'y'), 'y')", null),
+            ColumnUnknownException.class,
+            "The object `{x={y=10}}` does not contain the key `y`"
+        );
+        sqlExpressions.setErrorOnUnknownObjectKey(false);
+        assertEvaluate("subscript_obj(subscript_obj({x={y=10}}, 'y'), 'y')", null);
```

Comment:
I think it would be worth creating dedicated test cases for this with more descriptive names.